Upgrade-manager crashes if the ASG has 0 nodes

**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT

**What happened**:
- Create an ASG with 0 nodes.
- Create a CR for upgrading the ASG
- This should be a quick reconciliation. The controller should find that there are 0 nodes in the ASG and simply mark it as completed
- However, the controller crashes with the following error:
```
2020-07-01T07:56:15.449Z	DPANIC	controllers.RollingUpgrade	odd number of arguments passed as key-value pairs for logging	{"rollingupgrade": "instancemgr-20200701052602", "ignored key": "instancemgr-20200701052602"}
github.com/go-logr/zapr.handleFields
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:106
github.com/go-logr/zapr.(*infoLogger).Info
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:70
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).info
	/workspace/controllers/rollingupgrade_controller.go:1155
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).runRestack
	/workspace/controllers/rollingupgrade_controller.go:542
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).Process
	/workspace/controllers/rollingupgrade_controller.go:701
panic: odd number of arguments passed as key-value pairs for logging

goroutine 253 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000cd7b80, 0xc000ce9100, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/zapcore/entry.go:229 +0x545
go.uber.org/zap.(*Logger).DPanic(0xc000d922a0, 0x1b811d7, 0x3d, 0xc000ce9100, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:215 +0x7f
github.com/go-logr/zapr.handleFields(0xc000d922a0, 0xc000d90fc0, 0x1, 0x1, 0x0, 0x0, 0x0, 0xc000ce90c0, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:106 +0x5ce
github.com/go-logr/zapr.(*infoLogger).Info(0xc000d97c08, 0x1b456fb, 0x1d, 0xc000d90fc0, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:70 +0xb1
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).info(0xc0000b6000, 0xc0003fafc0, 0x1b456fb, 0x1d, 0xc000d90fc0, 0x1, 0x1)
	/workspace/controllers/rollingupgrade_controller.go:1155 +0xfe
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).runRestack(0xc0000b6000, 0xc000604860, 0xc0003fafc0, 0x1b3a4f0, 0x16, 0x0, 0x0, 0x0)
	/workspace/controllers/rollingupgrade_controller.go:542 +0x277
github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).Process(0xc0000b6000, 0xc000604860, 0xc0003fafc0)
	/workspace/controllers/rollingupgrade_controller.go:701 +0x82e
created by github.com/keikoproj/upgrade-manager/controllers.(*RollingUpgradeReconciler).Reconcile
	/workspace/controllers/rollingupgrade_controller.go:791 +0x745
```

**What you expected to happen**:
- ASGs with 0 nodes should be handled correctly.

**How to reproduce it (as minimally and precisely as possible)**:
See above.

**Anything else we need to know?**:

**Environment**:
- rolling-upgrade-controller version: 0.11

--- a/.semaphore/setup.sh
+++ b/.semaphore/setup.sh
@@ -7,7 +7,7 @@ export arch=$(go env GOARCH)
 
 # https://semaphoreci.com/docs/how-to-increase-the-amount-of-available-memory.html
 sudo swapoff -a
-sudo dd if=/dev/zero of=/swapfile bs=1M count=2048
+sudo dd if=/dev/zero of=/swapfile bs=1M count=4096
 sudo mkswap /swapfile
 sudo swapon /swapfile
 

--- a/controllers/rollingupgrade_controller.go
+++ b/controllers/rollingupgrade_controller.go
@@ -120,7 +120,7 @@ func (r *RollingUpgradeReconciler) runScript(script string, background bool, ruO
 		r.info(ruObj, "Running script in background. Logs not available.")
 		err := exec.Command(ShellBinary, "-c", script).Run()
 		if err != nil {
-			r.info(ruObj, "Script finished with error: %s", err)
+			r.info(ruObj, fmt.Sprintf("Script finished with error: %s", err))
 		}
 
 		return "", nil
@@ -535,11 +535,10 @@ func (r *RollingUpgradeReconciler) runRestack(ctx *context.Context, ruObj *upgra
 
 	r.info(ruObj, "Nodes in ASG that *might* need to be updated", "asgName", *asg.AutoScalingGroupName, "asgSize", len(asg.Instances))
 
-	// No further processing is required if ASG doesn't have an instance running
 	totalNodes := len(asg.Instances)
 	// No further processing is required if ASG doesn't have an instance running
 	if totalNodes == 0 {
-		r.info(ruObj, "Total nodes found for %s is 0", ruObj.Name)
+		r.info(ruObj, fmt.Sprintf("Total nodes needing update for %s is 0. Restack complete.", *asg.AutoScalingGroupName))
 		return 0, nil
 	}
 

